x-pack/platform/plugins/shared/lens/public/visualizations/datatable/visualization.tsx

```diff
@@ -126,14 +130,24 @@ export const getDatatableVisualization = ({
 
   triggers: [VIS_EVENT_TO_TRIGGER.filter, VIS_EVENT_TO_TRIGGER.tableRowContextMenuClick],
 
-  initialize(addNewLayer, state) {
-    return (
-      state || {
-        columns: [],
-        layerId: addNewLayer(),
-        layerType: LayerTypes.DATA,
-      }
-    );
+  initialize(addNewLayer, state, mainPalette, datasourceState) {
+    if (state) return convertToRuntimeState(state, datasourceState);
```

Comment:
In general, other than the comparison use case I think the deep copy is due

Reply:
Ok I added logic to clone the state except when used in `isEqual` vis method. See 1e920004c857f71ac7f02b069cc9728edf5aca28.

This does not apply to the xy vis as that state is always cloned via `convertPersistedState` prior to this PR.